internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -318,9 +215,67 @@ func (e *EnvStackConfig) StackName() string {
 	return NameForEnv(e.in.App.Name, e.in.Name)
 }
 
+// NewBootstrapEnvStackConfig sets up a BootstrapEnvStackConfig struct.
+func NewBootstrapEnvStackConfig(input *deploy.CreateEnvironmentInput) *BootstrapEnvStackConfig {
+	return &BootstrapEnvStackConfig{
+		in:     input,
+		parser: template.New(),
+	}
+}
+
+// BootstrapEnvStackConfig contains information for creating a stack bootstrapping environment resources.
+type BootstrapEnvStackConfig EnvStackConfig
+
+// Template returns the CloudFormation template to bootstrap environment resources.
+func (e *BootstrapEnvStackConfig) Template() (string, error) {
+	content, err := e.parser.ParseEnvBootstrap(&template.EnvOpts{
+		ArtifactBucketARN:    e.in.ArtifactBucketARN,
+		ArtifactBucketKeyARN: e.in.ArtifactBucketKeyARN,
+	})
+	if err != nil {
+		return "", err
+	}
+	return content.String(), nil
+}
+
+// Parameters returns the parameters to be passed into the bootstrap stack's CloudFormation template.
+func (e *BootstrapEnvStackConfig) Parameters() ([]*cloudformation.Parameter, error) {
+	return []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(envParamAppNameKey),
+			ParameterValue: aws.String(e.in.App.Name),
+		},
+		{
+			ParameterKey:   aws.String(envParamEnvNameKey),
+			ParameterValue: aws.String(e.in.Name),
+		},
+		{
+			ParameterKey:   aws.String(envParamToolsAccountPrincipalKey),
+			ParameterValue: aws.String(e.in.App.AccountPrincipalARN),
+		},
+	}, nil
+}
+
+// SerializedParameters returns the CloudFormation stack's parameters serialized
+// to a YAML document annotated with comments for readability to users.
+func (e *BootstrapEnvStackConfig) SerializedParameters() (string, error) {
+	// No-op for now.
+	return "", nil
+}
+
+// Tags returns the tags that should be applied to the bootstrap CloudFormation stack.
+func (e *BootstrapEnvStackConfig) Tags() []*cloudformation.Tag {
```

Comment:
Do we need to explicitly define these functions?

Reply:
Yes so that it implements `stackConfig`  interface.